provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -312,41 +306,49 @@ def _create_ports_descriptions(self, hostname: str) -> NetworkPorts:
         m0client_ports = []
         for srv in NetworkPorts.__annotations__.keys():
             if srv == 'hax':
-                url = conf.get('cortx>hare>hax>endpoints', allow_null=True)
+                num_ep = int(conf.get('cortx>hare>hax>num_endpoints'))
                 _hax = None
                 _hax_http = None
-                for u in url or ():
-                    _parsed_url = urlparse(u)
+                for i in range(num_ep):
+                    url = conf.get(f'cortx>hare>hax>endpoints[{i}]')
+                    _parsed_url = urlparse(url)
                     if _parsed_url.scheme in ('http', 'https'):
                         _hax_http = _parsed_url.port
                     else:
                         if _parsed_url.hostname == hostname:
                             _hax = round(_parsed_url.port / 100) * 100
             elif srv == 'm0_client_other':
-                for client in self.provider.get_motr_clients():
-                    url = client.get('endpoints')
-                    if url:
-                        for u in url or ():
-                            _parsed_url = urlparse(u)
+                num_clients = int(conf.get('cortx>motr>num_clients'))
+                for i in range(num_clients):
+                    num_endpoints = conf.get(
```

Comment:
Let's follow the same convention of getting the len of array.
```suggestion
                    num_endpoints = int(conf.get(
```